src/main/java/delete_event_use_case/DeleteEventInteractor.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Delete Event Interactor class.
+ * @author : Khizer Ahmad
+ * @created: November 19th 2022
+ * @since : 2022-11-28, Mon.
+ **/
+
+package delete_event_use_case;
+public class DeleteEventInteractor implements DeleteEventInputBoundary {
+
+    private DeleteEventOutputBoundary outputBoundary;
+    private DeleteEventDsGateway dsGateway;
+
+    public DeleteEventInteractor(
+            DeleteEventOutputBoundary outputBoundary, DeleteEventDsGateway dsGateway) {
+        this.outputBoundary = outputBoundary;
+        this.dsGateway = dsGateway;
+    }
+
+    @Override
+    public DeleteEventOutputData delete(DeleteEventInputData inputData) {
+        DeleteEventOutputData outputData = new DeleteEventOutputData(inputData.getDayIndex(), inputData.getEventTitle());
+
+        if (!dsGateway.eventExists(inputData.getDayIndex(), inputData.getEventTitle())) {
+            return outputBoundary.prepareFailureView(outputData, "The event does not exist.");
```

Comment:
This error message is very vague; explicitly state the name of the event that the user was going to delete. Besides, won't a user only be able to delete an event if it exists? The delete option will only show for events that are contained in the Day objects.

Reply:
As we are still at an intermediate part of our program being fully functional, I decided to leave possible error-checking for events, should anything faulty happen with our data persistence implementation. You are correct that technically there should never be an error with deleting an event, in the final product.